🧑‍🌾 Flaky `test_transport_py` failure in CI

## Description

Test regressions:
- projectroot.test_transport_py
- rosbag2_py.test.test_transport.test_recorder_api[sqlite3]
- rosbag2_py.test.test_transport.test_recorder_api[mcap]
- rosbag2_py.test.test_transport.test_player_unconfigured
- rosbag2_py.test.test_transport.test_recorder_unconfigured
- rosbag2_py.test.test_transport.test_record_cancel[sqlite3]
- rosbag2_py.test.test_transport.test_record_cancel[mcap]
- rosbag2_py.test.test_transport.test_play_cancel[sqlite3]
- rosbag2_py.test.test_transport.test_play_cancel[mcap]

Reference builds:
- [Jci__nightly-cyclonedds_ubuntu_noble_amd64#167](http://build.ros2.org/job/Jci__nightly-cyclonedds_ubuntu_noble_amd64/167)
- [Kci__nightly-connext_ubuntu_noble_amd64#29](http://build.ros2.org/job/Kci__nightly-connext_ubuntu_noble_amd64/29)
- [Kci__nightly-cyclonedds_ubuntu_noble_amd64#28](http://build.ros2.org/job/Kci__nightly-cyclonedds_ubuntu_noble_amd64/28)
- [Rci__nightly-cyclonedds_ubuntu_noble_amd64#473](http://build.ros2.org/job/Rci__nightly-cyclonedds_ubuntu_noble_amd64/473)
- [nightly_linux-aarch64_repeated#3250](https://ci.ros2.org/job/nightly_linux-aarch64_repeated/3250)
- [nightly_linux-rhel_repeated#2370](https://ci.ros2.org/job/nightly_linux-rhel_repeated/2370)
- [nightly_linux_repeated#3925](https://ci.ros2.org/job/nightly_linux_repeated/3925)
- [nightly_win_rep#3811](https://ci.ros2.org/job/nightly_win_rep/3811)

First failures started in 2025-07-28, I think one of the culprits might be https://github.com/ros2/rosbag2/pull/2062

<details><summary>Log output</summary>

First failure shows:

```
        metadata_io = rosbag2_py.MetadataIo()
        assert wait_for(
            lambda: metadata_io.metadata_file_exists(str(bag_path)),
            timeout=rclpy.duration.Duration(seconds=10))
        metadata = metadata_io.read_metadata(str(bag_path))
        bag_topics = [
            topic_info.topic_metadata.name
            for topic_info in metadata.topics_with_message_count
        ]
>       assert '/topic' in bag_topics, str(bag_topics)
E       AssertionError: []
E       assert '/topic' in []

../../src/ros2/rosbag2/rosbag2_py/test/test_transport.py:244: AssertionError
```

And then:

```
>       recorder = rosbag2_py.Recorder(
            storage_options, record_options, 'info', 'rosbag2_recorder_test')
E       RuntimeError: context is already initialized

../../src/ros2/rosbag2/rosbag2_py/test/test_transport.py:193: RuntimeError
___________________________ test_player_unconfigured ___________________________

    def test_player_unconfigured():
        # Test that using a constructor without full configuration raises an error when trying to use
        # the player
>       player = rosbag2_py.Player()
E       RuntimeError: context is already initialized

../../src/ros2/rosbag2/rosbag2_py/test/test_transport.py:250: RuntimeError
__________________________ test_recorder_unconfigured __________________________

    def test_recorder_unconfigured():
        # Test that using a constructor without full configuration raises an error when trying to use
        # the recorder
>       recorder = rosbag2_py.Recorder()
E       RuntimeError: context is already initialized

../../src/ros2/rosbag2/rosbag2_py/test/test_transport.py:278: RuntimeError

...
```

For the rest of the tests

</details>




### Expected Behavior
Tests should pass

### Actual Behavior
Tests are failing

## To Reproduce
1. Run a CI build in linux
2. See test regressions

## System (please complete the following information)
 - OS: Ubuntu, RHEL and Windows
 - ROS 2 Distro: Rolling, Kilted and Jazzy
 - Install Method: Source

--- a/rosbag2_py/test/test_transport.py
+++ b/rosbag2_py/test/test_transport.py
@@ -185,7 +185,7 @@ def test_recorder_api(tmp_path, storage_id):
 
     record_options = rosbag2_py.RecordOptions()
     record_options.start_paused = False
-    record_options.topics = ['/topic']
+    record_options.topics = ['/test_recorder_api_node/topic']
     record_options.is_discovery_disabled = False
     record_options.topic_polling_interval = datetime.timedelta(milliseconds=10)
     record_options.disable_keyboard_controls = True
@@ -201,7 +201,11 @@ def test_recorder_api(tmp_path, storage_id):
     ctx = rclpy.Context()
     ctx.init()
     node = rclpy.create_node('test_recorder_api_node', context=ctx)
-    pub = node.create_publisher(String, 'topic', QoSProfile(depth=10))
+    pub = node.create_publisher(String, '~/topic', QoSProfile(depth=10))
+    # Wait for the recorder to discover the topic
+    assert wait_for(
+        lambda: pub.get_subscription_count() > 0,
+        timeout=rclpy.duration.Duration(seconds=10))
     resume_client = node.create_client(Resume, '/rosbag2_recorder_test/resume')
     executor = rclpy.executors.SingleThreadedExecutor(context=ctx)
     executor.add_node(node)
@@ -241,7 +245,7 @@ def test_recorder_api(tmp_path, storage_id):
         topic_info.topic_metadata.name
         for topic_info in metadata.topics_with_message_count
     ]
-    assert '/topic' in bag_topics, str(bag_topics)
+    assert '/test_recorder_api_node/topic' in bag_topics, str(bag_topics)
 
 
 def test_player_unconfigured():

